nnvm/python/nnvm/frontend/tensorflow.py

```diff
@@ -0,0 +1,651 @@
+    'Sub'                               : _elemwise('sub'),
+    'Add'                               : _elemwise('add'),
+    'Rsqrt'                             : _rsqrt(),
+    'Squeeze'                           : _squeeze(),
+}
+
+
+class GraphProto(object):
+    """ A helper class for handling nnvm graph copying from Tensorflow GraphDef.
+    Definition:
+        https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/graph.proto
+    """
+    def __init__(self):
+        self._nodes = {}
+        self._params = {}
+        self._renames = {}
+        self._replacements = {}
+        self._output_shapes = {}
+        self._num_input = 0
+        self._num_param = 0
+        self._input_node = ''
+
+    def from_tensorflow(self, graph):
+        """Construct nnvm nodes from tensor flow  graph definition - GraphDef.
+
+        Follow the tensorflow graph definition to parse and convert it to NNVM.
+        Some of the assumptions listed below.
+
+            -> First Const or Placeholder node will be considered as graph input.
+            -> Rest all Const nodes are params.
+            -> Last node is assumed as graph output.
+            -> _output_shapes : Attribute should present in the tenserflow forzen graph.
+            -> DecodeJpeg, ResizeBilinear: These are dymmy operators.
+                                           Hence user should handle preprocessing outside.
+            -> CheckNumerics: No implementation as of now for this.
+                              Just copies input to output.
+
+
+        Parameters
+        ----------
+        graph : tensorflow graph definition object
+            The loaded tensorflow GraphDef
+
+        Returns
+        -------
+        sym : nnvm.sym.Symbol
+            The returned nnvm symbol
+        params : dict
+            A dict of name: tvm.nd.array pairs, used as pretrained weights
+        """
+        # Parse throught all nodes and start extracting
+        # params aka Const nodes
+        # input nodes  : First const node
+        # normal nodes : other normal nodes
+
+        try:
+            from tensorflow.python.framework import tensor_util
+        except ImportError as e:
+            raise ImportError(
+                "Unable to import tensorflow which is required {}".format(e))
+
+        for node in graph.node:
+            # Tensor flow doesn't have seperate list for params extraction.
+            # Operator name 'Const' is treated as a parameter to build NNVM params dict.
+            if node.op == "Placeholder":
+                # Assuming only one input graph with type 'Placeholder'
+                self._input_node = node.name
+                self._num_input += 1
+                self._nodes[node.name] = _sym.Variable(name=node.name)
+
+                self._output_shapes[node.name] = \
+                     [tensor_util.TensorShapeProtoToList(shape) \
+                     for shape in self._parse_attr(node.attr)['_output_shapes']]
```

Comment:
Does the code require a specific GraphDef version? 
When I try your feature, a problem troubles me. 
""
    for shape in self._parse_attr(node.attr)['_output_shapes']]
KeyError: '_output_shapes'
""

Reply:
Yes, you need to freeze the tensorflow graph with add_shapes=True like below.

output_graph_def = tf.graph_util.convert_variables_to_constants(
                    sess,
                    sess.graph.as_graph_def(add_shapes=True),
                    ['InceptionV3/Predictions/Reshape_1'],
                    )


It was discussed earlier @ https://github.com/dmlc/nnvm/pull/472

Reply:
thanks! "add_shapes=True"  can really avoid the error which happened on Placeholder, but the same error happens th on Const even if the fix is added.

Reply:
_output_shapes should exist for all nodes with above option.

Can you share the dump of the Const node ? Or Attribute dump of Const node ?

Reply:
"convert_variables_to_constants" method does not seem to add  "output_shape" field

Reply:
Hope you have added add_shapes option?

Can you share the model to check?

Reply:
I may not need the script. Just share the working protobuf. I will convert to have out shapes and check it.

Reply:
@colin1988 
From the script I see there are multiple inputs (Placeholders) for the graph.
Unfortunately I didn't consider multi input/output in this initial effort.

Reply:
@srkreddy1238 I want to cry! Looking forward to your updates

Reply:
I will try to share a patch for multi input later today :)

Can you share the protobuf file for reference ?

Reply:
@colin1988  
I just checked and multi input work fine. No need of any change.
Ref. added a test case for the same.

Reply:
@srkreddy1238 thanks very much. I will try it right now.